ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -778,45 +729,40 @@ static boolean eligible(BeaconState state, Validator validator) {
    */
   public static void finalUpdates(BeaconState state) throws EpochProcessingException {
     try {
-      UnsignedLong current_epoch = BeaconStateUtil.get_current_epoch(state);
-      UnsignedLong next_epoch = BeaconStateUtil.get_next_epoch(state);
-      UnsignedLong ENTRY_EXIT_DELAY = UnsignedLong.valueOf(Constants.ACTIVATION_EXIT_DELAY);
-      UnsignedLong LATEST_INDEX_ROOTS_LENGTH =
-          UnsignedLong.valueOf(Constants.LATEST_ACTIVE_INDEX_ROOTS_LENGTH);
-      UnsignedLong LATEST_RANDAO_MIXES_LENGTH =
-          UnsignedLong.valueOf(Constants.LATEST_RANDAO_MIXES_LENGTH);
-      UnsignedLong LATEST_PENALIZED_EXIT_LENGTH =
-          UnsignedLong.valueOf(Constants.LATEST_SLASHED_EXIT_LENGTH);
+      long current_epoch = BeaconStateUtil.get_current_epoch(state);
+      long next_epoch = BeaconStateUtil.get_next_epoch(state);
+      long ENTRY_EXIT_DELAY = Constants.ACTIVATION_EXIT_DELAY;
+      long LATEST_INDEX_ROOTS_LENGTH = Constants.LATEST_ACTIVE_INDEX_ROOTS_LENGTH;
+      long LATEST_RANDAO_MIXES_LENGTH = Constants.LATEST_RANDAO_MIXES_LENGTH;
+      long LATEST_PENALIZED_EXIT_LENGTH = Constants.LATEST_SLASHED_EXIT_LENGTH;
 
       // update hash tree root
-      int index = next_epoch.plus(ENTRY_EXIT_DELAY).mod(LATEST_INDEX_ROOTS_LENGTH).intValue();
+      long index = next_epoch + ENTRY_EXIT_DELAY % LATEST_INDEX_ROOTS_LENGTH;
```

Comment:
Another order of operations glitch. 
```suggestion
      long index = (next_epoch + ENTRY_EXIT_DELAY) % LATEST_INDEX_ROOTS_LENGTH;
```